### PR TITLE
More bug fixing related to index lists

### PR DESCRIPTION
**Pull request by simran-bhadani3**
*Catches the correct exceptions now
*Multiple spaces can be parsed correctly

### CODE REVIEW

File: src/main/java/seedu/cakecollate/logic/parser/exceptions/NegativeIndexException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package seedu.cakecollate.logic.parser.exceptions;
+
+
+/**
+ * Represents a parse error encountered by a parser, when the index provided is an integer above 32bits.
```
Comment (RuiXiong2211) [inline src/main/java/seedu/cakecollate/logic/parser/exceptions/NegativeIndexException.java:5]: I think this comment is incorrect lol, should be index provided is negative

File: src/test/java/seedu/cakecollate/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -50,6 +53,57 @@
...
+            "             1                   2          3              ";
+
+    private static final String INVALID_INDEX_LIST_WITH_ONE_INDEX = "-1";
+    private static final String INVALID_INDEX_LIST_WITH_TWO_INDEXES = "-1 -2";
+    private static final String INVALID_INDEX_LIST_WITH_THREE_INDEXES = "-1 -2 -3";
+    private static final String INVALID_INDEX_LIST_WITH_ONE_INDEX_AND_MULTIPLE_SPACES = "  -1  ";
+    private static final String INVALID_INDEX_LIST_WITH_TWO_INDEXES_AND_MULTIPLE_SPACES = "    -1    -2   ";
+    private static final String INVALID_INDEX_LIST_WITH_THREE_INDEXES_AND_MULTIPLE_SPACES = "   -1    -2    -3   ";
+    private static final String INVALID_INDEX_LIST_WITH_SINGLE_WHITESPACE = " ";
+    private static final String INVALID_INDEX_LIST_WITH_MULTIPLE_WHITESPACES = "     ";
```
Comment (RuiXiong2211) [inline src/test/java/seedu/cakecollate/logic/parser/ParserUtilTest.java:103]: think you can include more invalid test cases and varying the indexes where the error will pop up, and having duplicates 
eg, "-1 2 3"
eg " 1 1 1 1 "
eg "a 1 2"

Comment (simran-bhadani3) [reply src/test/java/seedu/cakecollate/logic/parser/ParserUtilTest.java:103]: Oh for index list I will be accepting duplicates because it is needed for some commands but will add in the other cases!

File: src/test/java/seedu/cakecollate/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -50,6 +53,57 @@
     private static final String VALID_TAG_2 = "neighbour";
     private static final String VALID_DELIVERY_DATE = "01/01/2022";
 
+    private static final String VALID_INDEX_LIST_WITH_ONE_INDEX = "1";
```
Comment (pPris) [inline src/test/java/seedu/cakecollate/logic/parser/ParserUtilTest.java:56]: Looks really extensive 👍 Maybe you could add in a test for commas in between the numbers + the numbers being out of order. Other than that lgtm
